bounded/src/set.rs

```diff
@@ -511,332 +608,267 @@ fn nullary_id_to_app(term: Term, relations: &[RelationDecl]) -> Term {
     }
 }
 
-/// A simplified syntax::Term that appears after quantifier enumeration.
-/// It structurally enforces negation normal form, and has logical semantics.
-#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub enum Ast {
-    // logic
-    /// A conjunction of terms. True is represented by And(set![]).
-    And(BTreeSet<Ast>),
-    /// A disjunction of terms. False is represented by Or(set![]).
-    Or(BTreeSet<Ast>),
-
-    // guards
-    /// An inclusion test
-    Includes(String, Vec<Element>), // r(x)
-    /// An exclusion test
-    Excludes(String, Vec<Element>), // !r(x)
-
-    // updates
-    /// An insertion
-    Insert(String, Vec<Element>), // r'(x)
-    /// A removal
-    Remove(String, Vec<Element>), // !r'(x)
-    /// A no-op (used for verifying that the module constrains all elements)
-    NoOp(String, Vec<Element>), // r'(x) = r(x)
-                                // r'(x) != r(x) could exist but isn't supported
-}
-
-impl Ast {
-    fn get_and(self) -> BTreeSet<Ast> {
-        match self {
-            Ast::And(terms) => terms,
-            _ => btreeset![self],
-        }
-    }
-
-    fn get_or(self) -> BTreeSet<Ast> {
-        match self {
-            Ast::Or(terms) => terms,
-            _ => btreeset![self],
-        }
-    }
-
-    fn get_guard(self, indices: &Indices) -> Result<Guard, TranslationError> {
-        match self {
-            Ast::Includes(set, element) => Ok(Guard {
-                index: indices[&(set.as_str(), element)],
-                value: true,
-            }),
-            Ast::Excludes(set, element) => Ok(Guard {
-                index: indices[&(set.as_str(), element)],
-                value: false,
-            }),
-            _ => Err(TranslationError::ExpectedGuard(self)),
+fn traverse_disjunction<T>(
+    term: &Term,
+    context: &Context,
+    assignments: &HashMap<String, Element>,
+    func: &impl Fn(&Term, &HashMap<String, Element>) -> Result<T, TranslationError>,
+) -> Result<Vec<T>, TranslationError> {
```

Comment:
Can you put a comment on this function describing what it does? I know it's not public, but I had trouble understanding it. It also does quantifier enumeration? What does it have to do with disjunction?